microsoft-edge/webview2/how-to/add-host-object-to-script.md

```diff
@@ -0,0 +1,295 @@
+---
+title: Use app objects in JavaScript
+description: Use host object in JavsScript using the WebView2 AddHostObjectToScript API.
+author: MSEdgeTeam
+ms.author: msedgedevrel
+ms.topic: conceptual
+ms.prod: microsoft-edge
+ms.technology: webview
+ms.date: 2/17/2022
+---
+# Use host objects in JavaScript
+
+WebView2 enables applications to bridge between web and native sides of an application using an object that can be passed between the two sides. Such objects are defined in the native code and often called host objects. They can be projected into JavaScript using a WebView2 API called `AddHostObjectToScript`, as described in this document.
+
+Why use `AddHostObjectToScript`?
+
+  * Developers often encounter a native object whose methods or properties they want to call from JavaScript. They may want these native object methods to be triggered by web code or by user interaction on the web side.
+
+  * In addition, developers may not want to re-implement native object functions, and this API enables re-use of code. For example, there might be a webcam native API, which would take a large amount of code to re-write on the web side. Having the ability to call the native object is quicker and more efficient than re-coding the function. In this case you can pass the object to the web side to reuse the native API function.
+
+Scenarios that may benefit from using host objects in script:
+
+  * If there is a keyboard API and the developer wants to call the `keyboardObject.showKeyboard` function from the web side.
+
+  * JavaScript is sandboxed, limiting its ability on the native side. For example, if you need to access a file on the native side, you must use the native file system. If you have a native object exposed to JavaScript via `AddHostObjectToScript`, you can use it to manipulate files on the native file system.
+
+This article uses the WebView2 sample app to demonstrate some practical applications of `AddHostObjectToScript`. For more information about how to embed web content into native applications [Embed web content into native applications - Microsoft Edge Development | Microsoft Docs](/microsoft-edge/webview2/how-to/communicate-btwn-web-native.md).
+
+
+<!-- ====================================================================== -->
+## Before you begin
```

Comment:
Change from major-step numbering scheme:
Before you begin
Step 1
Step 1a
Step 1b
Step 2
Step 3

to:

Step 1
Step 2
Step 3
Step 4
Step 5